test/test_common/environment.cc

```diff
@@ -158,7 +158,10 @@ void TestEnvironment::renameFile(const std::string& old_name, const std::string&
   // https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/rename-wrename?view=vs-2017
   // Note MoveFileEx cannot overwrite a directory as documented, nor a symlink, apparently.
   const BOOL rc = ::MoveFileEx(old_name.c_str(), new_name.c_str(), MOVEFILE_REPLACE_EXISTING);
-  ASSERT_NE(0, rc);
+  if (rc == 0) {
+    PANIC(fmt::format("failed to rename file from  {} to {} with error {}", old_name, new_name,
```

Comment:
Is this to ensure the test stops here (and doesn't continue on to time out)?

Reply:
yeah, it is quite annoying that the test times out if this rename fails. It clogs resources for no good reason

Reply:
huh, `ASSERT_*` should be fatal in google test: https://github.com/google/googletest/blob/master/googletest/docs/primer.md#assertions

Reply:
Interesting, I just verified it and you are right. I just updated it to include the error code in the assert.